### PR TITLE
Clarify /away error messages

### PR DESCRIPTION
**Pull request by whalemer**

### CODE REVIEW

File: server/chat-commands.js [hunk 1]
```diff
@@ -656,9 +656,10 @@ const commands = {
 
 	status(target, room, user, connection, cmd) {
 		if (!this.canTalk()) return;
-		if (!toID(target)) return this.parse('/help status');
+		if (!target) return this.parse('/help status');
+		if (!/[a-zA-Z0-9]/.test(target)) return this.errorReply("Your status must contain at least one letter or number.");
```
Comment (Zarel) [inline server/chat-commands.js]: Why?

Comment (whalemer) [reply server/chat-commands.js]: Which part are you asking about?

File: server/chat-commands.js [hunk 2]
```diff
@@ -676,8 +677,10 @@ const commands = {
 		if (!this.canTalk()) return;
 
 		let message = Chat.namefilter(target, user, true);
-		if (!message && toID(target)) return user.popup("Your status contains a banned word.");
-
+		if (target) {
+			if (!/[a-zA-Z0-9]/.test(target)) return this.errorReply("Your status must contain at least one letter or number.");
```
Comment (Zarel) [inline server/chat-commands.js]: What is this testing?

File: server/chat-commands.js [hunk 3]
```diff
@@ -698,11 +701,13 @@ const commands = {
 		} else {
 			awayType = `${awayType[0].toUpperCase()}${awayType.slice(1)}`;
 		}
-		if (toID(target)) {
+
+		if (target) {
+			if (!/[a-zA-Z0-9]/.test(target)) return this.errorReply("Your status must contain at least one letter or number.");
```
Comment (Zarel) [inline server/chat-commands.js]: Ditto

File: server/chat-commands.js (hunk 1 above)
Comment (Zarel) [reply server/chat-commands.js]: Why must statuses contain at least one letter or number?

**Comment by whalemer** [comment]
Statuses consisting only of symbols don't make it through the namefilter, because of
 ``	name = name.replace(/^[^A-Za-z0-9]+/, ""); // remove symbols from start`` 

This just catches that and gives a more friendly message.

File: server/chat-commands.js (hunk 1 above)
Comment (Zarel) [reply server/chat-commands.js]: Like, that's not actually an intended restriction; it was a restriction you seemingly accidentally added in #5577.

**Comment by Zarel** [comment]
Sounds like a problem with the namefilter.

**Comment by Zarel** [comment]
Although I do see the problem; yes, I would deal with this by fixing the namefilter.

**Comment by Zarel** [comment]
(This is a good next step in addition to fixing the namefilter)

**Pull request merged**